[BUG] Return value of `useHotkeys` incompatible with react 18

**Describe the bug**
When using the `ref` returned by `useHotkeys` on react <19, typescript complains that the type of `ReturnType<typeof useHotkeys<T>>` is incompatbile with `DetailedHTMLProps<HTMLAttributes<T>, T>["ref"]`.

**To Reproduce**

Create working directory:
```
$ mkdir ~/demo
$ cd ~/demo
```

Save the following to as `add-hotkey.patch` in your working directory:
```
diff --git a/src/App.tsx b/src/App.tsx
index 3d7ded3..cd3db74 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -1,14 +1,16 @@
 import { useState } from 'react'
+import { useHotkeys } from 'react-hotkeys-hook'
 import reactLogo from './assets/react.svg'
 import viteLogo from '/vite.svg'
 import './App.css'
 
 function App() {
   const [count, setCount] = useState(0)
+  const ref = useHotkeys<HTMLDivElement>("a", () => { setCount((c) => c + 1) }, []);
 
   return (
     <>
-      <div>
+      <div ref={ref}>
         <a href="https://vite.dev" target="_blank">
           <img src={viteLogo} className="logo" alt="Vite logo" />
         </a>
```

Test react v19, builds without issue
```
$ cd ~/demo
$ npm create vite@6.3.1 -- react19 --template react-ts
$ cd react19
$ npm install
$ npm install react-hotkeys-hook@5.0.1
$ git init
$ git add .
$ git commit -m "initial"
$ git apply ../add-hotkey.patch
$ npm run build
```

Test react v18, build fails
```
$ cd ~/demo
$ npm create vite@6.1.1 -- react18 --template react-ts
$ cd react18
$ npm install
$ npm install react-hotkeys-hook@5.0.1
$ git init
$ git add .
$ git commit -m "initial"
$ git apply ../add-hotkey.patch
$ npm run build
```

Following error is displayed:
```
$ npm run build                       

> react18@0.0.0 build
> tsc -b && vite build

src/App.tsx:13:12 - error TS2322: Type 'RefObject<RefType<HTMLDivElement>>' is not assignable to type 'LegacyRef<HTMLDivElement> | undefined'.
  Type 'RefObject<RefType<HTMLDivElement>>' is not assignable to type 'RefObject<HTMLDivElement>'.
    Type 'RefType<HTMLDivElement>' is not assignable to type 'HTMLDivElement'.
      Type 'null' is not assignable to type 'HTMLDivElement'.

13       <div ref={ref}>
              ~~~

  node_modules/@types/react/index.d.ts:303:9
    303         ref?: LegacyRef<T> | undefined;
                ~~~
    The expected type comes from property 'ref' which is declared here on type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>'


Found 1 error.
```

sandbox reproducing error with react 18: https://codesandbox.io/p/sandbox/8n4xws

**Expected behavior**
Should be compatible with react 18 and react 19

**Desktop (please complete the following information):**
 - OS: all
 - Browser: n/a
 - Version: 5.0.1

--- a/packages/react-hotkeys-hook/src/lib/types.ts
+++ b/packages/react-hotkeys-hook/src/lib/types.ts
@@ -4,8 +4,6 @@ export type FormTags = 'input' | 'textarea' | 'select' | 'INPUT' | 'TEXTAREA' | 
 export type Keys = string | readonly string[]
 export type Scopes = string | readonly string[]
 
-export type RefType<T> = T | null
-
 export type EventListenerOptions =
   | {
   capture?: boolean

--- a/packages/react-hotkeys-hook/src/lib/useHotkeys.ts
+++ b/packages/react-hotkeys-hook/src/lib/useHotkeys.ts
@@ -1,4 +1,4 @@
-import { HotkeyCallback, Keys, Options, OptionsOrDependencyArray, RefType } from './types'
+import { HotkeyCallback, Keys, Options, OptionsOrDependencyArray } from './types'
 import { DependencyList, useCallback, useEffect, useLayoutEffect, useRef } from 'react'
 import { mapCode, parseHotkey, parseKeysHookInput } from './parseHotkeys'
 import {
@@ -28,7 +28,7 @@ export default function useHotkeys<T extends HTMLElement>(
   options?: OptionsOrDependencyArray,
   dependencies?: OptionsOrDependencyArray
 ) {
-  const ref = useRef<RefType<T>>(null)
+  const ref = useRef<T>(null)
   const hasTriggeredRef = useRef(false)
 
   const _options: Options | undefined = !(options instanceof Array)

